common/djangoapps/cors_csrf/middleware.py

```diff
@@ -65,3 +131,78 @@ def is_secure_patched():
         res = super(CorsCSRFMiddleware, self).process_view(request, callback, callback_args, callback_kwargs)
         request.is_secure = is_secure_default
         return res
+
+
+class CsrfCrossDomainCookieMiddleware(object):
+    """Set an additional "cross-domain" CSRF cookie.
+
+    Usage:
+
+        1) Decorate a view with `@ensure_csrf_cookie_cross_domain`.
+        2) Set `CROSS_DOMAIN_CSRF_COOKIE_NAME` and `CROSS_DOMAIN_CSRF_COOKIE_DOMAIN`
+            in settings.
+        3) Add the domain to `CORS_ORIGIN_WHITELIST`
+        4) Enable `FEATURES['ENABLE_CROSS_DOMAIN_CSRF_COOKIE']`
+
+    For testing, it is often easier to relax the security checks by setting:
+        * `CORS_ALLOW_INSECURE = True`
+        * `CORS_ORIGIN_ALLOW_ALL = True`
+
+    """
+
+    def __init__(self):
+        """Disable the middleware if the feature is not enabled. """
+        if not settings.FEATURES.get('ENABLE_CROSS_DOMAIN_CSRF_COOKIE'):
+            log.debug("CsrfCrossDomainCookieMiddleware has been disabled.")
+            raise MiddlewareNotUsed()
+
+        if not getattr(settings, 'CROSS_DOMAIN_CSRF_COOKIE_NAME', ''):
+            raise ImproperlyConfigured(
+                "You must set `CROSS_DOMAIN_CSRF_COOKIE_NAME` when "
+                "`FEATURES['ENABLE_CROSS_DOMAIN_CSRF_COOKIE']` is True."
+            )
+
+        if not getattr(settings, 'CROSS_DOMAIN_CSRF_COOKIE_DOMAIN', ''):
+            raise ImproperlyConfigured(
+                "You must set `CROSS_DOMAIN_CSRF_COOKIE_DOMAIN` when "
+                "`FEATURES['ENABLE_CROSS_DOMAIN_CSRF_COOKIE']` is True."
+            )
+
+    def process_response(self, request, response):
+        """Set the cross-domain CSRF cookie. """
+
+        # Check whether this is a secure request from a domain on our whitelist.
+        if not is_cross_domain_request_allowed(request):
+            log.debug("Could not set cross-domain CSRF cookie.")
+            return response
+
+        # Check whether (a) the CSRF middleware has already set a cookie, and
+        # (b) this is a view decorated with `@ensure_cross_domain_csrf_cookie`
+        # If so, we can send the cross-domain CSRF cookie.
+        should_set_cookie = (
+            request.META.get('CROSS_DOMAIN_CSRF_COOKIE_USED', False) and
+            request.META.get('CSRF_COOKIE_USED', False) and
+            request.META.get('CSRF_COOKIE') is not None
+        )
+
+        if should_set_cookie:
+            # This is very similar to the code in Django's CSRF middleware
+            # implementation, with two exceptions:
+            # 1) We change the cookie name and domain so it can be used cross-domain.
+            # 2) We always set "secure" to True, so that the CSRF token must be
+            # sent over a secure connection.
+            response.set_cookie(
+                settings.CROSS_DOMAIN_CSRF_COOKIE_NAME,
+                request.META['CSRF_COOKIE'],
+                max_age=(60 * 60 * 24 * 7 * 52),
```

Comment:
Django 1.4 hard-codes the same value in its CSRF middleware.  Django 1.7, introduced the `CSRF_COOKIE_AGE` setting to make it configurable.

Maybe it would make sense to add `CSRF_COOKIE_AGE` as a setting with this value, for forwards compatibility with future versions of Django?